tools/compiletest/src/runtest.rs

```diff
@@ -387,6 +392,99 @@ impl<'test> TestCx<'test> {
+            adapted_lines.push(line.clone());
+            if adapted_lines.last().unwrap() == "RESULTS:" {
+                break;
+            }
+        }
+
+        if line_iter.clone().peekable().peek().is_none() {
+            // iterated over all the lines, so this is not the new format
+            return output.to_string();
+        }
+
+        // 2. Adapt output of the form:
+        //
+        //        Check 2: foo.assertion.2
+        //        - Status: SUCCESS
+        //        - Description: "assertion failed: x == 2"
+        //
+        //    to:
+        //
+        //       [foo.assertion.2] line X assertion failed: x == 2: SUCCESS
+        //
+        // (Note that since the line numbers are missing from the new format, it
+        // will be printed as "line X")
+        let mut check_index = 1usize;
+        while let Some(line) = line_iter.next() {
+            // Check line is of the form: "Check <n>: foo.assertion.5"
+            if let Some(name) = line.strip_prefix(format!("Check {}: ", check_index).as_str()) {
+                // Result line is of the form: "\t - Status: <Result>"
+                let result = line_iter.next().unwrap().strip_prefix("\t - Status: ").unwrap();
+                // description could span multiple lines, e.g.
+                // - Description: "Code generation sanity check: Correct CBMC vtable size for StructTag("tag-_5528490532708175949") (MIR type Closure(DefId(2:9423 ~ std[c344]::rt::lang_start::{closure#0}), [(), i8, extern "rust-call" fn(()) -> i32, (fn(),)])). Please report failures:
+                // https://github.com/model-checking/kani/issues/new?template=bug_report.md"
+                // Extract all the lines to the next empty line which marks the end of the description
+                let mut description: Vec<String> = Vec::new();
+                while let Some(line) = line_iter.next() {
+                    if line.is_empty() {
+                        break;
+                    }
+                    description.push(line.clone());
+                }
+                assert!(!description.is_empty());
+
+                // Strip the '\t - Description: "' (including double quotes) from
+                // the beginning (first line) of the description
+                *description.first_mut().unwrap() = description
+                    .first()
+                    .unwrap()
+                    .strip_prefix("\t - Description: \"")
+                    .unwrap()
+                    .to_string();
+                // Strip the double quotes from the end (last line) of the description
+                *description.last_mut().unwrap() =
+                    description.last().unwrap().strip_suffix("\"").unwrap().to_string();
+
+                // Change the output to the following form:
+                //     [<assertion name>] line X <description>: <result>
+                // where description could have multiple lines
```

Comment:
Just checking. The description itself includes explicit line breaks, right? We don't wrap the lines.